main/src/cgeo/geocaching/Image.java

```diff
@@ -15,27 +17,92 @@
 
 import java.io.File;
 
+/**
+ * Represent an Image along with Title and Description.
+ * Mostly used for representing image in/from logs.
+ */
 public class Image implements Parcelable {
-    private final String url;
-    private final String title;
-    private final String description;
+    @NonNull private Uri uri;
```

Comment:
Just a small off-topic note: You mean 'pierre angulaire' here? I think this should rather be a 'cornerstone' then.